src/v2/Utils/Hooks/__tests__/useDialog.jest.tsx

```diff
@@ -0,0 +1,50 @@
+import React from "react"
+import { renderHook, act } from "@testing-library/react-hooks"
+import { render, screen } from "@testing-library/react"
+import { useDialog } from "../useDialog"
+
+const onShow = jest.fn()
+const onHide = jest.fn()
+const hookProps = { Dialog: () => <div>Dialog</div>, onShow, onHide }
+
+describe("useDialog", () => {
+  afterEach(() => jest.clearAllMocks())
+
+  it("returns with visible false and dialog null", async () => {
+    const { result } = renderHook(() => useDialog(hookProps))
+    render(result.current.dialogComponent)
+
+    expect(result.current.isVisible).toBeFalsy()
+    expect(screen.queryByText("Dialog")).not.toBeInTheDocument()
+  })
+
+  it("visibility is set to true", () => {
+    const { result, waitForNextUpdate } = renderHook(() => useDialog(hookProps))
+
+    act(() => {
+      result.current.showDialog()
+    })
+    waitForNextUpdate()
```

Comment:
This function [returns a promise](https://react-hooks-testing-library.com/reference/api#waitfornextupdate), right? 

Do we not need to `await waitForNextUpdate()`? (Noticed this in the [official example](https://react-hooks-testing-library.com/usage/advanced-hooks#async))

Reply:
Good point, I needed because ran into some flakiness, but from my previous experience with the module never needed it, very rare occasions. Removed 👍